blockchain/genesis.go

```diff
@@ -307,16 +308,25 @@ func (g *Genesis) ToBlock(baseStateRoot common.Hash, db database.DBManager) *typ
 		db = database.NewMemoryDBManager()
 	}
 	stateDB, _ := state.New(baseStateRoot, state.NewDatabase(db), nil, nil)
+	rules := params.Rules{}
+	if g.Config != nil {
+		rules = g.Config.Rules(new(big.Int).SetUint64(g.Number))
+	}
 	for addr, account := range g.Alloc {
-		if len(account.Code) != 0 {
-			originalCode := stateDB.GetCode(addr)
+		originalCode := stateDB.GetCode(addr)
+		if _, ok := types.ParseDelegation(account.Code); ok && rules.IsPrague {
+			stateDB.SetCodeToEOA(addr, account.Code, rules)
+		} else if len(account.Code) == 0 && len(account.Storage) != 0 && rules.IsPrague {
+			stateDB.CreateEOA(addr, false, accountkey.NewAccountKeyLegacy())
+		} else if len(account.Code) != 0 {
+			stateDB.CreateSmartContractAccount(addr, params.CodeFormatEVM, rules)
 			stateDB.SetCode(addr, account.Code)
-			// If originalCode is not nil,
-			// just update the code and don't change the other states
-			if originalCode != nil {
-				logger.Warn("this address already has a not nil code, now the code of this address has been changed", "addr", addr.String())
-				continue
-			}
+		}
+		// If account.Code is nil and originalCode is not nil,
+		// just update the code and don't change the other states
+		if len(account.Code) != 0 && originalCode != nil {
+			logger.Warn("this address already has a not nil code, now the code of this address has been changed", "addr", addr.String())
+			continue
```

Comment:
Looks like this can be checked first before L316-322.

Reply:
I can't do that. I want to do continue after setting code even if an original code exists.